src/main/java/org/folio/linked/data/integration/kafka/sender/search/AuthorityCreateMessageSender.java

```diff
@@ -59,11 +59,11 @@ private Set<Resource> findAuthorities(Resource resource, Set<Resource> visitedRe
   }
 
   private boolean isNewAndNotVisited(Resource resource, Set<Resource> visited) {
-    return resource.isNew() && !visited.contains(resource);
+    return resource.getIndexDate() == null && !visited.contains(resource);
   }
 
   private boolean isNewAuthority(Resource resource) {
-    return resource.isNew() && isAuthority(resource);
+    return resource.getIndexDate() == null && isAuthority(resource);
```

Comment:
Cover this changes with a unit test please

Reply:
done